pkg/storage/store.go

```diff
@@ -99,6 +105,86 @@ func (s *store) LazyQuery(ctx context.Context, req logql.SelectParams) (iter.Ent
 			lazyChunks = append(lazyChunks, &chunkenc.LazyChunk{Chunk: c, Fetcher: fetchers[i]})
 		}
 	}
+	return lazyChunks, nil
+}
+
+func (s *store) GetSeries(ctx context.Context, req logql.SelectParams) ([]logproto.SeriesIdentifier, error) {
+	matchers, _, from, through, err := decodeReq(req)
+	if err != nil {
+		return nil, err
+	}
+
+	lazyChunks, err := s.lazyChunks(ctx, matchers, from, through)
+	if err != nil {
+		return nil, err
+	}
+
+	// group chunks by series
+	chunksBySeries := partitionBySeriesChunks(lazyChunks)
+
+	firstChunksPerSeries := make([]*chunkenc.LazyChunk, 0, len(chunksBySeries))
+
+	// discard all but one chunk per series
+	for _, chks := range chunksBySeries {
+		firstChunksPerSeries = append(firstChunksPerSeries, chks[0][0])
+	}
+
+	results := make([]logproto.SeriesIdentifier, 0, len(firstChunksPerSeries))
+
+	// bound concurrency
+	groups := make([][]*chunkenc.LazyChunk, 0, len(firstChunksPerSeries)/s.cfg.MaxChunkBatchSize+1)
+
+	split := s.cfg.MaxChunkBatchSize
+	if len(firstChunksPerSeries) < split {
+		split = len(firstChunksPerSeries)
```

Comment:
Can you cover that line please ?

Reply:
should be covered by 
https://github.com/grafana/loki/pull/1914/files#diff-7b012faf877ce550ea7c9baaace4ee6bR396

which sets the batch size in tests to 1